tests/test_client.py

```diff
@@ -452,6 +452,15 @@ def test_request_alive_check(mock_socket):
     assert mock_socket.tx_queue[-1] == alive_check_request
 
 
+def test_alive_check(mock_socket):
+    sut = DoIPClient(test_ip, test_logical_address)
+    mock_socket.rx_queue.append(alive_check_request)
+    with pytest.raises(TimeoutError):
```

Comment:
I wonder what does this test do that test_activation_with_alive_check doesn't.

Reply:
`test_alive_check()` only appends `alive_check_request` to the rx queue, whereas `test_activation_with_alive_check()` appends a further message (`successful_activation_response`) after the `alive_check_request`.

During the while loop of `read_doip()` the first message (`alive_check_request`) was read from the rx queue but got stuck in the `Parser`, so `Parser.read_message()` returned `None`.
Then the next message (`successful_activation_response`) was read from the queue, handed over to `Parser.read_message()`, which now returned the first message ... finally test passed.